readthedocs/projects/tasks/builds.py

```diff
@@ -900,7 +899,7 @@ def store_build_artifacts(self):
                 # Re-raise the exception to fail the build and handle it
                 # automatically at `on_failure`.
                 # It will clearly communicate the error to the user.
-                raise BuildAppError("Error uploading files to the storage.")
+                raise BuildAppError("Error uploading files to the storage.") from exc
```

Comment:
Oh neat, I've never seen this pattern used before. I like this, it's explicit :+1:

Reply:
It was a suggestion from the new version of prospector 😉 . I also liked because it's more explicit 👍🏼